runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -1017,21 +1294,229 @@ MM_SchedulingDelegate::calculateEdenSize(MM_EnvironmentVLHGC *env)
 	Trc_MM_SchedulingDelegate_calculateEdenSize_Exit(env->getLanguageVMThread(), (_edenRegionCount * regionSize));
 }
 
-UDATA
+void
+MM_SchedulingDelegate::moveTowardRecommendedEden(MM_EnvironmentVLHGC *env, double edenChangeSpeed) 
+{
+	Assert_MM_true(edenChangeSpeed <= 1 && edenChangeSpeed >= 0);
+
+	if ((0 == _historicalPartialGCTime) || (0 == _averagePgcInterval)) {
+		/* Until we have collected any information about PGC time, we don't have the data we need to make informed decision about eden size  */
+		return;
+	}
+
+	uintptr_t recommendedEdenSizeBytes = calculateRecommendedEdenSize(env);
```

Comment:
While the code is very well documented, I would be shy to expand names of moveTowardRecommendedEden, calculateRecommendedEdenSize.... with ForExpandedHeap